File: order.service/src/main/java/br/com/edu/order/infra/rest/v1/response/ItemResponse.java
/* (C)2024 */
package br.com.edu.order.infra.rest.v1.response;

import br.com.edu.order.domain.Item;
import java.math.BigDecimal;
import java.util.UUID;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class ItemResponse {

    private UUID id;

    private String product;

    private BigDecimal price;

    private Integer quantity;


    public static ItemResponse fromModel(final Item item) {
        return ItemResponse.builder()
                .id(item.getId())
                .product(item.getProduct())
                .price(item.getPrice())
                .quantity(item.getQuantity())
                .build();
    }
}

File: order.service/order-service-app/src/main/kotlin/order/infra/rest/v1/response/ItemResponse.kt
package order.infra.rest.v1.response

import com.fasterxml.jackson.annotation.JsonInclude
import order.domain.Item
import java.math.BigDecimal
import java.util.*

@JsonInclude(JsonInclude.Include.NON_NULL)
data class ItemResponse(
    val id: UUID? = null,

    val product: String? = null,

    val price: BigDecimal? = null,

    val quantity: Int? = null
) {

    companion object {
        fun fromModel(item: Item) = ItemResponse(
            id = item.id,
            product = item.product,
            price = item.price,
            quantity = item.quantity
        )
    }
}
